src/temporal/t.rast.aggregate.seasons/t.rast.aggregate.seasons.py

```diff
@@ -0,0 +1,242 @@
+            remod.flags.d = True
+            remod.flags.quiet = True
+            remod.run()
+
+
+def main():
+    options, flags = gs.parser()
+    strds = options["input"]
+
+    output_name = options["output"]
+
+    tgis.init()
+    # We need a database interface
+    dbif = tgis.SQLDatabaseInterfaceConnection()
+    dbif.connect()
+    mapset = tgis.core.get_current_mapset()
+
+    if options["years"] != "":
+        try:
+            vals = options["years"].split("-")
+            years = range(vals)
+        except:
+            try:
+                years = options["years"].split(",")
+            except:
+                gs.fatal(_("Invalid option years"))
+    else:
+        if strds.find("@") >= 0:
+            id_ = strds
+        else:
+            id_ = f'{strds}@{gs.gisenv()["MAPSET"]}'
+        dataset = tgis.dataset_factory("strds", id_)
+        dataset.select(dbif)
+        ext = dataset.get_temporal_extent()
+        years = range(ext.start_time.year, ext.end_time.year)
+
+    method = options["method"]
+    basename = options["basename"]
+    nprocs = int(options["nprocs"])
+    register_null = flags["n"]
+
+    seasons = ["spring", "summer", "autumn", "winter"]
+
+    # create new space time vector datasets one for each year to be used as sampler
+    for year in years:
+        season_vect = []
+        for seas in seasons:
+            name = f"sample_{seas}_{year}"
+            vect = VectorTopo(name)
+            vect.open("w")
+            point = Point(0, 0)
+            vect.write(point, cat=1)
+            vect.close()
+            map_layer = tgis.space_time_datasets.VectorDataset(f"{name}@{mapset}")
+            if seas == "spring":
+                extent = tgis.AbsoluteTemporalExtent(
+                    start_time=datetime(int(year), 3, 20),
+                    end_time=datetime(int(year), 6, 21),
+                )
+            elif seas == "summer":
+                extent = tgis.AbsoluteTemporalExtent(
+                    start_time=datetime(int(year), 6, 21),
+                    end_time=datetime(int(year), 9, 20),
+                )
+            elif seas == "autumn":
+                extent = tgis.AbsoluteTemporalExtent(
+                    start_time=datetime(int(year), 9, 20),
+                    end_time=datetime(int(year), 12, 21),
+                )
+            elif seas == "winter":
+                extent = tgis.AbsoluteTemporalExtent(
+                    start_time=datetime(int(year), 12, 21),
+                    end_time=datetime(int(year) + 1, 3, 20),
+                )
+            map_layer.set_temporal_extent(extent=extent)
+            season_vect.append(map_layer)
+        temp_season = f"sample_seasons_{year}"
+        outsp = tgis.open_new_stds(
+            temp_season,
+            "stvds",
+            "absolute",
+            f"Season vector year {year}",
+            f"Season vector for the year {year}",
+            "mean",
+            dbif,
+            gs.overwrite(),
```

Comment:
I would expect that the global overwrite would be respected here, but I'm not sure from top of my head.

```suggestion
```